packages/amplication-code-gen-types/src/plugins-types.ts

```diff
@@ -0,0 +1,133 @@
+import { AppInfo, DTOs, Entity, Module, Role } from "./code-gen-types";
+import winston from "winston";
+
+export interface PrintResultType {
+  code: string;
+  map?: any;
+  toString(): string;
+}
+
+export interface ContextUtil {
+  setDsgPath: (path: string) => string;
+  skipDefaultBehavior: boolean;
+  importStaticModules: (source: string, basePath: string) => Promise<Module[]>;
+}
+export interface DsgContext {
+  appInfo: AppInfo;
+  entities: Entity[];
+  roles: Role[];
+  modules: Module[];
+  DTOs: DTOs;
+  plugins: PluginMap;
+  logger: winston.Logger;
+  utils: ContextUtil;
+}
+
+export type PluginWrapper = (args: any[], func: () => void) => any;
+
+export interface EventParams {}
+
+export interface DsgPlugin {
+  id?: string;
+  name?: string;
+  description?: string;
+  packageName: string;
+}
+
+export type PluginMap = {
+  [K in EventNames]?: {
+    before?: (<T>(context: DsgContext, params: any) => T)[];
+    after?: (<T>(context: DsgContext, params: any) => T)[];
+  };
+};
+
+export interface CreateServiceModulesParams extends EventParams {
+  before: {
+    entityName: string;
+    entityType: string;
+    entity: Entity;
+    srcDirectory: string;
+    extraMapping: { [key: string]: any };
+  };
+  after: Module[];
+}
+
+export interface CreateControllerModulesParams extends EventParams {
+  before: {
+    resource: string;
+    entityName: string;
+    entityType: string;
+    entityServiceModule: string;
+    entity: Entity;
+    srcDirectory: string;
+    extraMapping: { [key: string]: any };
+  };
+  after: Module[];
+}
```

Comment:
because we wil have so many of these, it is better to use a separate file just for the events